apps/condo/domains/meter/schema/PropertyMeter.test.js

```diff
@@ -0,0 +1,283 @@
+
+        describe('hard delete', () => {
+            test('admin can\'t', async () => {
+                await expectToThrowAccessDeniedErrorToObj(async () => {
+                    await PropertyMeter.delete(admin, 'id')
+                })
+            })
+
+            test('user can\'t', async () => {
+                const client = await makeClientWithNewRegisteredAndLoggedInUser()
+                await expectToThrowAccessDeniedErrorToObj(async () => {
+                    await PropertyMeter.delete(client, 'id')
+                })
+            })
+
+            test('anonymous can\'t', async () => {
+                const client = await makeClient()
+                await expectToThrowAccessDeniedErrorToObj(async () => {
+                    await PropertyMeter.delete(client, 'id')
+                })
+            })
+        })
+
+        describe('read', () => {
+            test('admin can', async () => {
+                const [organization] = await createTestOrganization(admin)
+                const [property] = await createTestProperty(admin, organization)
+                const [meter] = await createTestPropertyMeter(admin, organization, property, resource, {})
+                const objs = await PropertyMeter.getAll(admin, {}, { sortBy: ['updatedAt_DESC'] })
+
+                expect(objs.length).toBeGreaterThanOrEqual(1)
+                expect(objs).toEqual(expect.arrayContaining([
+                    expect.objectContaining({
+                        id: meter.id,
+                    }),
+                ]))
+            })
+
+            test('user can', async () => {
+                const client = await makeClientWithNewRegisteredAndLoggedInUser()
+
+                const [organization] = await createTestOrganization(admin)
+                const [property] = await createTestProperty(admin, organization, { map: buildingMapJson })
+                const [role] = await createTestOrganizationEmployeeRole(admin, organization, {})
+                await createTestOrganizationEmployee(admin, organization, client.user, role)
+
+                const [objCreated] = await createTestPropertyMeter(client, organization, property, resource)
+
+                const objs = await PropertyMeter.getAll(client, {}, { sortBy: ['updatedAt_DESC'] })
+
+                expect(objs).toHaveLength(1)
+                expect(objs[0]).toMatchObject({
+                    id: objCreated.id,
+                })
+            })
+
+            test('anonymous can\'t', async () => {
+                const client = await makeClient()
+                await expectToThrowAuthenticationErrorToObjects(async () => {
+                    await PropertyMeter.getAll(client, {}, { sortBy: ['updatedAt_DESC'] })
+                })
+            })
+        })
+    })
+
+    describe('Validation tests', () => {
+
+        test('employee with "canManageMeters" role: cannot create Meter if Meter with same number exist in user organization', async () => {
```

Comment:
It will be easier to read validation tests, if description of test cases will start right from the validation rule, for example:
```suggestion
        test('uniqueness within organization', async () => {
```

If we will omit the code, it will be read naturally as "test uniqueness within organization".

You can merge two test cases into one, no need to split.
Take a look at apps/condo/domains/banking/schema/BankTransaction.test.js:583, — first create operation is executed successfully without violating constraints, second fails.